cdap-app-fabric/src/test/java/io/cdap/cdap/internal/operation/OperationLifecycleManagerTest.java

```diff
@@ -0,0 +1,192 @@
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.Mockito;
+
+public class OperationLifecycleManagerTest extends OperationTestBase {
+  protected static TransactionRunner transactionRunner;
+  private static final String testNamespace = "test";
+  private static OperationLifecycleManager operationLifecycleManager;
+  private static int batchSize;
+
+  @ClassRule public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
+
+  private static EmbeddedPostgres postgres;
+
+  @Before
+  public void before() throws Exception {
+    TransactionRunners.run(
+        transactionRunner,
+        context -> {
+          OperationRunStore operationRunsStore = new OperationRunStore(context);
+          operationRunsStore.clearData();
+        });
+  }
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    CConfiguration cConf = CConfiguration.create();
+    postgres = PostgresInstantiator.createAndStart(cConf, TEMP_FOLDER.newFolder());
+    Injector injector =
+        Guice.createInjector(
+            new ConfigModule(cConf),
+            new LocalLocationModule(),
+            new SystemDatasetRuntimeModule().getInMemoryModules(),
+            new StorageModule(),
+            new AbstractModule() {
+              @Override
+              protected void configure() {
+                bind(MetricsCollectionService.class)
+                    .to(NoOpMetricsCollectionService.class)
+                    .in(Scopes.SINGLETON);
+              }
+            });
+
+    transactionRunner = injector.getInstance(TransactionRunner.class);
+    operationLifecycleManager =
+        new OperationLifecycleManager(transactionRunner, Mockito.mock(OperationRuntime.class));
+    StoreDefinition.OperationRunsStore.create(injector.getInstance(StructuredTableAdmin.class));
+    batchSize = cConf.getInt(AppFabric.STREAMING_BATCH_SIZE);
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    Closeables.closeQuietly(postgres);
+  }
+
+  @Test
+  public void testScanOperations() throws Exception {
+    List<OperationRunDetail> insertedRuns = insertTestRuns(transactionRunner);
+    // get a filtered list of testNamespace runs
+    List<OperationRunDetail> testNamespaceRuns =
+        insertedRuns.stream()
+            .filter(detail -> detail.getRunId().getNamespace().equals(testNamespace))
+            .collect(Collectors.toList());
+
+    TransactionRunners.run(
+        transactionRunner,
+        context -> {
+          List<OperationRunDetail> gotRuns = new ArrayList<>();
+          List<OperationRunDetail> expectedRuns;
+          ScanOperationRunsRequest request;
+
+          // verify the scan without filters picks all runs for testNamespace
+          request = ScanOperationRunsRequest.builder().setNamespace(testNamespace).build();
+          operationLifecycleManager.scanOperations(request, batchSize, d -> gotRuns.add(d));
```

Comment:
replace `d -> gotRuns.add(d)` to `gitRuns:add`

Reply:
done